src/main/java/com/alex/eshop/service/CategoryService.java

```diff
@@ -27,6 +29,25 @@ public List<CategoryDTO> getAllCategories(Pageable pageable) {
     }
 
     public CategoryDTO getCategory(Long id) {
-        return categoryMapper.toDto(categoryRepository.findById(id).orElseThrow(() -> new DataNotFound("There is no category with id " + id)));
+        return categoryMapper.toDto(categoryRepository.findById(id)
+                .orElseThrow(()->new DataNotFound("There is no category with id " + id)));
+    }
+    public CategoryDTO createCategory(CategoryCreateDTO categoryCreateDTO){
+       return categoryMapper.toDto(categoryRepository.save(categoryMapper.toEntity(categoryCreateDTO)));
+    }
+    public CategoryDTO updateCategory(CategoryUpdateDTO categoryUpdateDTO){
+        if(!categoryRepository.existsById(categoryUpdateDTO.getId())){
+            throw new DataNotFound("There is no category with id " + categoryUpdateDTO.getId());
+        }
+        return categoryMapper.toDto(categoryRepository.save(categoryMapper.toEntity(categoryUpdateDTO)));
+       }
+    public void deleteCategory(Long id){
+        if(!categoryRepository.existsById(id)){
+            throw new DataNotFound("There is no category with id " + id);
+        }
+        if(!(categoryRepository.countCategoryByItems(id) == 0)) {
```

Comment:
the comparison is too complicated. simpler to use `categoryRepository.countCategoryByItems(id) > 0`